docs/src/use/configure/migration-guide.md

```diff
@@ -0,0 +1,442 @@
+
+export default [
+    js.configs.recommended,
+    {
+        ...customTestConfig,
+        files: ["**/*.test.js"],
+    },
+];
+```
+
+### Ignoring Files
+
+With eslintrc, you can make ESLint ignore files by creating a separate `.eslintignore` file in the root of your project. The `.eslintignore` file uses the same glob pattern syntax as `.gitignore` files. Alternatively, you can use an `ignorePatterns` property in your eslintrc file.
+
+To ignore files with flat config, you can use the `ignores` property in a config object. The `ignores` property accepts an array of glob patterns. Note that flat config glob patterns do _not_ match dot files (e.g. `.dotfile.js`).
+
+For example, here's a `.eslintignore` example you can use with an eslintrc config:
+
+```shell
+# .eslintignore
+temp.js
+.config/*
+# ...other ignored files
+```
+
+`ignorePatterns` example:
+
+```javascript
+// .eslintrc.js
+module.exports = {
+    // ...other config
+    ignorePatterns: ["temp.js", ".config/*"],
+};
+```
+
+Here are the same files ignore patterns in flat config:
+
+```javascript
+export default [
+    {
+        // ...other config
+        ignores: ["temp.js", ".config/*"]
+    }
+];
+```
+
+### Linter Options
+
+ESlintrc files let you configure the linter itself with the `noInlineConfig` and `reportUnusedDisableDirectives` properties.
+
+The flat config system introduces a new top-level property `linterOptions` that you can use to configure the linter. In the `linterOptions` object, you can include `noInlineConfig` and `reportUnusedDisableDirectives`.
+
+For example, here's an eslintrc file with linter options enabled:
+
+```javascript
+// .eslintrc.js
+
+module.exports = {
+    // ...other config
+    noInlineConfig: true,
+    reportUnusedDisableDirectives: true
+}
+```
+
+Here's the same options in flat config:
+
+```javascript
+// eslint.config.js
+
+export default [
+    {
+        // ...other config
+        linterOptions: {
+            noInlineConfig: true,
+            reportUnusedDisableDirectives: true
+        }
+    }
+];
+```
+
+### CLI Flag Changes
+
+The following CLI flags are no longer supported with the flat config file format:
+
+* `--rulesdir`
+* `--ext`
+* `--resolve-plugins-relative-to`
+
+The flag `--no-eslintrc` has been replaced with `--no-config-lookup`.
+
+### Additional Changes
+
+The following changes have been made from the eslintrc to the flat config file format:
+
+* The `root` option no longer exists. (Flat config files act as if `root: true` is set.)
+* The `files` option cannot be a single string anymore, it must be an array.
+* The `sourceType` option now supports the new value `"commonjs"` (`.eslintrc` supports it too, but it was never documented).
+
+## TypeScript Types for Flat Config Files
+
+You can see the TypeScript types for the eslint.config.js file format in the DefinitelyTyped project. The interface for the objects in the config’s array is called the `FlatConfig`.
```

Comment:
```suggestion
You can see the TypeScript types for the `eslint.config.js` file format in the DefinitelyTyped project. The interface for the objects in the config’s array is called the `FlatConfig`.
```

Shall we use backticks here for consistency with the other filenames in the document?

Reply:
```suggestion
You can see the TypeScript types for the flat config file format in the DefinitelyTyped project. The interface for the objects in the config’s array is called the `FlatConfig`.
```